src/nodes.coffee

```diff
@@ -936,15 +936,30 @@ exports.NumberLiteral = class NumberLiteral extends Literal
         @parsedValue = @value
         @value = "#{@value}"
       else
-        @parsedValue = Number @value
+        @parsedValue = parseNumber @value
 
-  astType: -> 'NumericLiteral'
+  isBigInt: ->
+    /n$/.test @value
+
+  astType: ->
+    if @isBigInt()
+      'BigIntLiteral'
+    else
+      'NumericLiteral'
 
   astProperties: ->
     return
-      value: @parsedValue
+      value:
+        if @isBigInt()
+          @parsedValue.toString()
```

Comment:
...and the `value`/`extra.rawValue` for BigInt's is the quoted value (eg `"24"` for the source `"24n"`) in Babel AST's